Legacy/bonej/src/main/java/org/bonej/plugins/ParticleDisplay.java

```diff
@@ -425,6 +440,40 @@ static void display3DOriginal(final ImagePlus imp, final int resampling, final I
 
 	// ----------------- HELPER METHODS --------------//
 
+	/**
+	 * Calculate a colour based on particle parameters
+	 * 
+	 * @param p particle number, for GRADIENT
+	 * @param nSurfaces total number of particles being displayed for GRADIENT
+	 * @param colourMode SPLIT, GRADIENT or ORIENTATION
+	 * @param volumes particle volumes, for SPLIT
+	 * @param eigens particle eigendecompositions, for ORIENTATION
+	 * @param splitValue volumt at which colour changes
+	 * 
+	 * @return RGB (Color3f) colour of the particle
+	 */
+	private static Color3f getColour(int p, int nSurfaces, int colourMode, double[] volumes, 
+			EigenvalueDecomposition[] eigens, double splitValue) {
+		Color3f colour = new Color3f(0, 0, 0);
+		if (colourMode == GRADIENT) {
+			final float red = 1.0f - p / (float) nSurfaces;
+			final float green = 1.0f - red;
+			final float blue = p / (2.0f * nSurfaces);
+			colour = new Color3f(red, green, blue);
+		} else if (colourMode == SPLIT) {
+			if (volumes[p] > splitValue) {
+				// red if over
+				colour = new Color3f(1.0f, 0.0f, 0.0f);
+			} else {
+				// yellow if under
+				colour = new Color3f(1.0f, 1.0f, 0.0f);
+			}
+		} else if (colourMode == ORIENTATION) {
+			colour = ParticleDisplay.colourFromEigenVector(eigens[p]);
+		}
+		return colour;
```

Comment:
In principle, if we knew that colour modes might change, a way to defend against coding errors here would be
```
private enum ColourMode = {GRADIENT, SPLIT, ORIENTATION};
...
else if (colourMode == ORIENTATION) {
  return ParticleDisplay.colourFromEigenVector(eigens[p]);
}

throw new AssertionError("Code should not reach here - not all colour modes are covered");
```

Even if black was a valid default colour, then `DEFAULT` should be one additional enum value and `else if` clause.